@here/harp-mapview/lib/ClipPlanesEvaluator.ts

```diff
@@ -0,0 +1,524 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EarthConstants, MathUtils, Projection, ProjectionType } from "@here/harp-geoutils";
+import { assert } from "@here/harp-utils";
+import * as THREE from "three";
+
+export interface ClipPlanesEvaluator {
+    /**
+     * Compute near and far clipping planes distance.
+     *
+     * Evaluation method should be called on every frame  and camera clip planes updated.
+     * This is related to evaluator implementation and its input data, that may suddenly change
+     * such as camera position or angle, projection type or so.
+     * Some evaluators may not depend on all or even any of input objects, but to preserve
+     * compatibility with any evaluator type it is strongly reccomended to update on every frame.
+     * @param camera The [[THREE.Camera]] object used in scene which clipping planes are evaluated.
+     * @param projection The current [[Projection]] method.
+     * @note Camera clipping planes should be not automatically updated via #evaluateClipPlanes()
+     * call, user should do it manually if needed.
+     */
+    evaluateClipPlanes(camera: THREE.Camera, projection: Projection): { near: number; far: number };
+}
+
+/**
+ * Simple camera clip planes evaluator, interpolates near and far planes based on ground distance.
+ *
+ * This evaluator supports both planar and spherical projections, although it behaviour is
+ * slightly different in each case. General algorithm setups near plane in between camera and
+ * ground level, while far plane is just calculated using scale and bias approach with far offset
+ * and multiplier.
```

Comment:
Add also that this doesn't take into consideration whether an elevation provider has been set up.